### ISSUE TITLE
Incorrect regex in UsPhoneRecognizer matches "ddd-343-\dddd"

### ISSUE THREAD

**Issue by efung** [microsoft/presidio#638]
**Describe the bug**
While trying to implement my own phone recognizer, I was studying the patterns in `UsPhoneRecognizer` and noticed that the `Phone (strong)` pattern has a few typos. It causes the recognizer to return a score of 0.7 on the input `ddd-343-\dddd`

In addition, a number such as `212-555-1212` is recognized as a phone number with only a score of 0.5, which is arguably too low.

**To Reproduce**
1. Set up Presidio in a Python shell with an analyzer.
2. Execute `analyzer.analyze(text="ddd-343-\dddd", language='en', entities=['PHONE_NUMBER'])`
3. Observe the output is `[type: PHONE_NUMBER, start: 0, end: 13, score: 0.7]`
4. Execute `analyzer.analyze(text="212-555-1212", language='en', entities=['PHONE_NUMBER'])`
5. Observe the output is `[type: PHONE_NUMBER, start: 0, end: 12, score: 0.5]`

**Expected behavior**
I would not expect the first input above to be recognized as a valid phone number. And I would expect the second input to score higher.

**Additional context**
The `Phone (strong)` pattern has a few typos. It should be:

```diff
@ presidio-analyzer/presidio_analyzer/predefined_recognizers/us_phone_recognizer.py:19 @ class UsPhoneRecognizer(PatternRecognizer):
     PATTERNS = [
         Pattern(
             "Phone (strong)",
-            r"(\(\d{3}\)\s*\d{3}[-\.\s]??\d{4}|d{3}[-\.\s]\d{3}[-\.\s]\\d{4})",
+            r"(\(\d{3}\)\s*\d{3}[-\.\s]??\d{4}|\d{3}[-\.\s]\d{3}[-\.\s]\d{4})",
```

Moreover, with the fix above, an input like `212-555-1212` would match both the `Phone (strong)` and `Phone (medium)` patterns.

It looks like the intent of the medium phone pattern is for when an area code is not present. However, when an area code is present, because of the bug, a ten-digit number isn't matching the strong phone pattern.

**Comment by omri374** [comment]
Thanks @efung! Would you be interested in creating a PR with a fix?

**Comment by efung** [comment]
> Thanks @efung! Would you be interested in creating a PR with a fix?

Sure.  Can you comment on what conceptually is represented by the strong vs medium regexes?  Which pattern should match when the input contains a number like `201-555-1212`?

Also, do you need a signed contributor license agreement or anything like that?

**Comment by navalev** [comment]
Hi @efung, most contributions require you to agree to a Contributor License Agreement (CLA) declaring that you have the right to, and actually do, grant us the rights to use your contribution. For details, visit https://cla.microsoft.com.
When you submit a pull request, a CLA-bot will automatically determine whether you need to provide a CLA

**Comment by navalev** [comment]
Resolved in #658. Closing this issue
